curvlinops/kfac.py

```diff
@@ -164,11 +183,24 @@ def __init__(
                 f"Invalid loss_average: {loss_average}. "
                 f"Must be 'batch' or 'batch+sequence' if loss_func.reduction != 'sum'."
             )
+        if loss_func.reduction == "sum" and loss_average is not None:
+            # reduction used in loss function will overwrite loss_average
+            warnings.warn(
+                f"Loss function uses reduction='sum', but loss_average={loss_average}."
+                " loss_average is set to None.",
+                stacklevel=2,
+            )
+            loss_average = None
```

Comment:
I also thought about this, but I think in practice it is extremely likely that the user simply wants to use the reduction specified in the loss function and just forgot to adjust the `loss_average` argument. Ideally, we would set the loss function reduction based on the `loss_average` argument (so there can't be a mismatch), but this would change the base API which is why I didn't implement it.

But I'm open to raise an exception here as well, what do you think?

Reply:
I would prefer the more strict version (exception). We can always relax it later if we want.